sql/date-and-time-types.md

```diff
@@ -0,0 +1,227 @@
+
+When dealing with date and time value types, please note: 
+
++ Although TiDB tries to interpret different formats, the date-portion must be in the format of year-month-day (for example, '1998-09-04'), rather than month-day-year or day-month-year.
++ If the year-portion of a date is specified as 2 digits, TiDB converts it based on the following rules:
+  - Values between 70 and 99 is converted to a value between 1970 and 1999
+  - Values between 00 and 69 is converted to a value between 2000 and 2069
++ If a numberic value is needed in the context, TiDB automatically converts the date or time value into a numeric type. For example,
+
+  ```sql
+  mysql> SELECT NOW(), NOW()+0, NOW(3)+0;
+  +---------------------+----------------+--------------------+
+  | NOW()               | NOW()+0        | NOW(3)+0           |
+  +---------------------+----------------+--------------------+
+  | 2012-08-15 09:28:00 | 20120815092800 | 20120815092800.889 |
+  +---------------------+----------------+--------------------+
+  ```
+  
++ TiDB may automatically convert invalid values or values beyond the supported range to a zero value of that type. This behavior is dependent on the SQL Mode set. For example,
+
+  ```sql
+  mysql> set @@sql_mode = '';                               
+  Query OK, 0 rows affected (0.00 sec)
+
+  mysql> show create table t1;
+  +-------+---------------------------------------------------------------------------------------------------------+
+  | Table | Create Table                                                                                            |
+  +-------+---------------------------------------------------------------------------------------------------------+
+  | t1    | CREATE TABLE `t1` (
+    `a` time DEFAULT NULL
+  ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+  +-------+---------------------------------------------------------------------------------------------------------+
+  1 row in set (0.00 sec)
+  
+  mysql> insert into t1 values ('2090-11-32:22:33:44');
+  Query OK, 1 row affected, 1 warning (0.00 sec)
+  
+  mysql> select * from t1;
+  +----------+
+  | a        |
+  +----------+
+  | 00:00:00 |
+  +----------+
+  1 row in set (0.00 sec)
+  ```
+
++ Setting different SQL modes can change TiDB behaviors.
++ TiDB allows month or day in the columns of `DATE` and `DATETIME` to be zero value, for example, '2009-00-00' or '2009-01-00'. If this date type is to be calculated in a function, for example, in `DATE_SUB()` or `DATE_ADD()`, the result can be incorrect. 
++ TiDB allows zero value '0000-00-00' to be stored. Sometimes '0000-00-00' is more convenient for users than NULL value if they are not using the `NO_ZERO_DATE` SQL mode.
```

Comment:
```suggestion
+ By default, TiDB enables the SQL mode `NO_ZERO_DATE`. This mode prevents storing zero values such as '0000-00-00'.
```
I think we will receive criticism from standards geeks if we say '0000-00-00' is more convenient :-)  This is similar to saying "sometimes lying is more convenient than the truth".